ddtrace/tracer/option.go

```diff
@@ -177,7 +177,7 @@ func forEachStringTag(str string, fn func(key string, val string)) {
 func newConfig(opts ...StartOption) *config {
 	c := new(config)
 	c.sampler = NewAllSampler()
-	c.agentAddr = resolveAgentAddr(defaultAddress)
+	c.agentAddr = resolveAgentAddr("")
```

Comment:
It's odd with the empty string. I understand why you did it, but I think it would be better to move the env. var. code below with the rest of the env. var. code. Basically out of the `resolveAgentAddr` function.